Integrate Portfolio page with Netlify CMS data

mecies opened the pull request:
## Description

Setup for reading data from markdown in page components and PortfolioPage with data read from markdown.

Display from markdown files:
- [x] Portfolio page
- [x] Developer profile

This PR covers:
- Removing stories for Portfolio page`s (they were causing annoying errors after GraphQL setup) and they are pretty redundant after the page is done
- Displaying data on the `portfolio-page` from markdown files
- Fixing `PortfolioProjectDialog` component to accept missing props
- Adding `dayjs` library to format dates
- Fixing height in CardDesktop component 

Closes #67 

## Preview

### Desktop
<img width="1391" alt="Screenshot 2020-09-04 at 21 31 35" src="https://user-images.githubusercontent.com/39158460/92278746-25b1db80-eef6-11ea-856f-e41d6d239bd4.png">
<img width="1388" alt="Screenshot 2020-09-04 at 21 31 40" src="https://user-images.githubusercontent.com/39158460/92278747-26e30880-eef6-11ea-9ec4-522d1ac0a155.png">
<img width="727" alt="Screenshot 2020-09-04 at 21 31 49" src="https://user-images.githubusercontent.com/39158460/92278752-28accc00-eef6-11ea-88af-da6ad4481187.png">
<img width="733" alt="Screenshot 2020-09-04 at 21 31 54" src="https://user-images.githubusercontent.com/39158460/92278756-2a768f80-eef6-11ea-9052-9146b192f80f.png">

### Tablet
<img width="1015" alt="Screenshot 2020-09-04 at 21 33 20" src="https://user-images.githubusercontent.com/39158460/92278864-64e02c80-eef6-11ea-8e1e-2b2331d7d820.png">
<img width="781" alt="Screenshot 2020-09-04 at 21 33 32" src="https://user-images.githubusercontent.com/39158460/92278876-6ad60d80-eef6-11ea-91bc-c90385c2df8a.png">
<img width="857" alt="Screenshot 2020-09-04 at 21 33 37" src="https://user-images.githubusercontent.com/39158460/92278882-6c9fd100-eef6-11ea-947c-d5519dd82540.png">

### Mobile
<img width="314" alt="Screenshot 2020-09-04 at 21 33 53" src="https://user-images.githubusercontent.com/39158460/92278900-745f7580-eef6-11ea-9470-912c8f506150.png">
<img width="313" alt="Screenshot 2020-09-04 at 21 34 02" src="https://user-images.githubusercontent.com/39158460/92278904-775a6600-eef6-11ea-8d51-2df4c0309a87.png">
<img width="313" alt="Screenshot 2020-09-04 at 21 34 07" src="https://user-images.githubusercontent.com/39158460/92278907-79242980-eef6-11ea-84e7-45c72947e83d.png">

mecies committed:
Setup for graphql with markdown with portfolio page in progress

mecies committed:
Display information on portfolio page from graphql

mecies committed:
Add dayjs library to format dates

mecies committed:
Remove portfolio project story

mecies committed:
Fixes to portfolio project component

mecies committed:
Add content to portfolio-page index.md file

mecies committed:
Fix height for preview note in CardDesktop component

mecies committed:
Add PortfolioProjectDialog component to portfolio page

mecies committed:
Use destructuring in PortfolioProjectDialog

mecies committed:
Change hardcoded image to regex

mecies committed:
Add different image for project for preview purpose

mecies committed:
Name fix

mecies committed:
Change fetching image to regular URL string

mecies committed:
Add missing resume download link

KlaraGajaszek commented on src/pages/portfolio-page/index.tsx:
```diff
@@ -153,3 +203,53 @@ const PortfolioPage = () => {
+        }
+      }
+    }
+    developerProfile: markdownRemark(fileAbsolutePath: { regex: "/developer-profile/index-1.md/" }) {
```
DetailsCard perform in more than one page component, so maybe you can move this 
query to other query or use useStaticQuery directly in the DetailsCard component and then everybody will use the same query and we won't duplicate our code ?

mecies replied:
Yeah, I think that it's a great idea, I will try to use it directly in component 👍

mecies committed:
Fetch developer profile in DetailsCard component

mecies committed:
Remove redundant mapping util

mecies committed:
Fix DetailsCard stories

mecies reviewed (commented):
I have an error related to storybook and usage of static query and not sure how to fix it
https://www.gatsbyjs.com/docs/visual-testing-with-storybook/

KonradSzwarc commented on src/components/PortfolioProject/PortfolioProjectDesktop.tsx:
```diff
@@ -59,24 +59,24 @@ const useStyles = makeStyles((theme) => ({
 }));
 
-const BlogPostDialogDesktop: FC<PortfolioProjectDialogProps> = ({ isOpen, ...props }) => {
+const BlogPostDialogDesktop: FC<PortfolioProjectDialogProps> = ({ ...props }) => {
```
You don't need to destructure props here

KonradSzwarc commented on src/components/PortfolioProject/PortfolioProject.stories.tsx:
```diff
@@ -20,6 +23,10 @@ const basePropsData = {
   contentMainRole:
     'Sed eu enim auctor, auctor ipsum id, finibus metus. Donec vel efficitur mi. Donec at justo nunc. Nullam vitae urna ut leo vehicula sodales. Vestibulum ut dui laoreet, consequat mi a, aliquam quam. Vivamus ac felis non lectus dapibus pulvinar. Nulla imperdiet maximus mi vel lacinia. Sed sagittis velit sed arcu hendrerit sollicitudin. Quisque faucibus bibendum egestas.',
+  tags: [{ name: 'Redux' }, { name: 'React' }, { name: 'TypeScript' }],
+  handleClose: () => window.alert('closed'),
```
As it's a Storybook maybe using an action would be a better choice for logging events?

KonradSzwarc commented on src/components/PortfolioProject/PortfolioProjectTablet.tsx:
```diff
@@ -60,24 +60,24 @@ const useStyles = makeStyles((theme) => ({
 }));
 
-const BlogPostDialogTablet: FC<PortfolioProjectDialogProps> = ({ isOpen, ...props }) => {
+const BlogPostDialogTablet: FC<PortfolioProjectDialogProps> = ({ ...props }) => {
```
No need for destructuring

mecies committed:
CR fixes

Removed redundant props destructuring in PortfolioProjectCard component and replaced dummy window alerts with action in storybook file for PortfolioProjectCard

KonradSzwarc commented on src/components/PortfolioProject/PortfolioProject.stories.tsx:
```diff
@@ -20,6 +24,10 @@ const basePropsData = {
   contentMainRole:
     'Sed eu enim auctor, auctor ipsum id, finibus metus. Donec vel efficitur mi. Donec at justo nunc. Nullam vitae urna ut leo vehicula sodales. Vestibulum ut dui laoreet, consequat mi a, aliquam quam. Vivamus ac felis non lectus dapibus pulvinar. Nulla imperdiet maximus mi vel lacinia. Sed sagittis velit sed arcu hendrerit sollicitudin. Quisque faucibus bibendum egestas.',
+  tags: [{ name: 'Redux' }, { name: 'React' }, { name: 'TypeScript' }],
+  handleClose: action('onClick'),
```
3 times onClick? 😉

mecies replied:
Sorrry for being lazy 🤣

mecies committed:
Fix PortfolioProjectCard storybook action names

mecies committed:
Fix static query not working in storybook

mecies committed:
Merge branch 'master' into feat/portfolio-page-with-cms

mecies committed:
Fix developer profile content file

mecies committed:
Move resume page types to views directory

mecies committed:
Move blog page files to views directory

mecies committed:
Fix about page import and content file

mecies committed:
Remove details card component

mecies committed:
Remove page stories
